worlds/pokemon_crystal/phone.py

```diff
@@ -0,0 +1,58 @@
+from typing import TYPE_CHECKING
+
+from BaseClasses import ItemClassification
+from .data import data
+from .phone_data import get_shuffled_basic_calls, template_call_bike_shop, template_call_psychic, template_call_remote
+
+if TYPE_CHECKING:
+    from . import PokemonCrystalWorld
+
+
+def generate_phone_traps(world: "PokemonCrystalWorld"):
+    if world.options.phone_trap_weight.value:
```

Comment:
This `if` isn't necessary here, since the same condition is used to decide whether to call this function in the first place.

If that weren't the case, there are a couple variables that get initialized inside this `if` block which are used below it, which would cause exceptions if this condition were ever `False` while in this function.